topics/releases.md

```diff
@@ -1,144 +1,98 @@
+### Patch versions
 
-Patches primarily consist of bug fixes and very rarely introduce any
-compatibility issues.
+PATCH versions are released with backwards compatible bug fixes and should not introduce new features.
 
-Upgrading from a previous patch-level version is almost always safe and
-seamless.
+Upgrading from a previous patch version should be safe and seamless.
+It should be safe to run a Valkey cluster with servers running on different patch versions.
 
-New features and configuration directives may be added, or default values
-changed, as long as these don’t carry significant impacts or introduce
-operations-related issues.
+PATCH versions may also introduce small improvements such as performance or memory optimizations that don't come with any tradeoffs.
 
 ### Minor versions
 
-Minor versions usually deliver maturity and extended functionality.
+MINOR version are released with new functionality that is added in a backward compatible manner.
+Examples of new functionality include new commands, info fields, or configuration parameters.
+
+Upgrading from a previous minor version should be safe, and will not introduce incompatibilities between servers in the cluster.
 
-Upgrading between minor versions does not introduce any application-level
-compatibility issues.
+**NOTE:** Minor releases may include new commands and data types that can introduce incompatibility between servers in the cluster, but users need to opt-in to these features to cause this type of incompatibility.
+For this reason, it is not recommended to run a Valkey cluster with servers running on different minor versions.
+Users should also avoid new features until all servers in the cluster have been upgrades.
 
-Minor releases may include new commands and data types that introduce
-operations-related incompatibilities, including changes in data persistence
-format and replication protocol.
+Commands may also be deprecated in minor versions.
+If a command is deprecated, a replacement command or an alternative to using the command will be defined in the same minor version.
 
 ### Major versions
 
-Major versions introduce new capabilities and significant changes.
+MAJOR versions are released with significant functionality that may break backwards compatibility or alter key performance characteristics.
+Examples of significant functionality include altering the behavior of an existing command, removing previously deprecated commands, changing the default value of configs, and significant refactoring for performance improvements.
 
-Ideally, these don't introduce application-level compatibility issues.
+Upgrading from a previous major version is intended to be safe, but should be approached with caution. 
+You should carefully read the release notes before performing a major version upgrade.
+Major version upgrades do not guarantee backwards compatibility, which means you should always upgrade replicas before upgrading primaries in order to ensure data consistency.
```

Comment:
How to do rolling upgrade in a cluster or sentinel is described somewhere in the docs, right? Can we link to it?

Reply:
I couldn't find an example of it. It's on the list of stuff we should document better. https://github.com/valkey-io/valkey-doc/issues/115